Library/Homebrew/cmd/fetch.rb

```diff
@@ -32,6 +32,11 @@ def fetch
   def fetch_bottle? f
     return true if ARGV.force_bottle? && f.bottle
     return false unless f.bottle && f.pour_bottle?
+    if formula.file_modified?
+      ohai "#{formula.full_name}.rb is modified; fetching source"
```

Comment:
`formula.path.to_s.gsub("#{HOMEBREW_PREFIX}/", '')` might be a bit better (particularly for stuff from taps).